std/algorithm/searching.d

```diff
@@ -1361,7 +1361,20 @@ if (isInputRange!InputRange &&
 
     alias EType  = ElementType!R;
 
-    static if (isNarrowString!R)
+    // If the haystack is a SortedRange we can use binary search to find the needle.
+    // Works only for the default find predicate and any SortedRange predicate.
+    // 8829 enhancement
+    import std.range: SortedRange;
+    static if (is(InputRange : SortedRange!TT, TT) && isDefaultPred)
+    {
+        auto lb = haystack.lowerBound(needle);
+        if ((lb.length == 0 && haystack[0] != needle) || lb.length == haystack.length
+           || haystack[lb.length] != needle)
+            return haystack[$ .. $];
```

Comment:
Odd alignment - first, there's one character difference which is definitely a problem (all indentation levels should be a multiple of 4). Second, code is (approximately) at the same level as an unrelated expression continuation, which can't be right.